src/api/entities/Proposal/index.ts

```diff
@@ -147,25 +147,46 @@ export class Proposal extends Entity<UniqueIdentifiers> {
    * Retrieve the proposal details
    */
   public async getDetails(): Promise<ProposalDetails> {
+    const { context, pipId } = this;
+
     const {
-      context: {
-        polymeshApi: {
-          query: { pips },
+      data: {
+        proposal: {
+          pipId: id,
+          proposer,
+          createdAt,
+          url,
+          description,
+          coolOffEndBlock,
+          endBlock,
+          proposal: rawProposal,
+          lastState,
+          lastStateUpdatedAt,
+          totalVotes,
+          totalAyesWeight,
+          totalNaysWeight,
         },
       },
-      pipId,
-    } = this;
-
-    const rawProposal = await pips.proposals(pipId);
-    const {
-      state,
-      proposal: { sectionName, methodName },
-    } = rawProposal.unwrap();
+    } = await context.queryMiddleware<Ensured<Query, 'proposal'>>(
+      proposal({
+        pipId,
+      })
+    );
 
     return {
-      state: meshProposalStateToProposalState(state),
-      module: sectionName,
-      method: methodName,
+      pipId: id,
+      proposer,
+      createdAt,
+      url,
+      description,
+      coolOffEndBlock,
+      endBlock,
+      call: rawProposal ? meshProposalToCall(rawProposal, context) : undefined,
```

Comment:
Why wouldn't `rawProposal` be defined?

Reply:
Proposals that uses `setCode` are returned as null by the harvester